lisp/bazel-build.el

```diff
@@ -24,38 +24,56 @@
 
 ;;; Code:
 
+(require 'subr-x)
+
 (require 'bazel-util)
 
+(defgroup bazel-build nil
+  "Package for editing, building, and running code using Bazel."
+  :link '(url-link "https://bazel.build")
+  :link '(url-link "https://github.com/bazelbuild/emacs-bazel-mode")
+  :group 'languages)
+
+(defcustom bazel-build-bazel-command '("bazel")
+  "Command and arguments that should be used to invoke Bazel."
+  :type '(repeat string)
+  :risky t
+  :group 'bazel-build)
+
 (defun bazel-build (target)
   "Build a Bazel TARGET."
-  (interactive (list (bazel-build--read-target "bazel build ")))
+  (interactive (list (bazel-build--read-target "build")))
   (bazel-build--run-bazel-command "build" target))
 
 (defun bazel-run (target)
   "Build and run a Bazel TARGET."
-  (interactive (list (bazel-build--read-target "bazel run ")))
+  (interactive (list (bazel-build--read-target "run")))
   (bazel-build--run-bazel-command "run" target))
 
 (defun bazel-test (target)
   "Build and run a Bazel test TARGET."
-  (interactive (list (bazel-build--read-target "bazel test ")))
+  (interactive (list (bazel-build--read-target "test")))
   (bazel-build--run-bazel-command "test" target))
 
 (defun bazel-build--run-bazel-command (command target)
   "Run Bazel tool with given COMMAND, e.g. build or run, on the given TARGET."
   (compile
-   (mapconcat #'shell-quote-argument (list "bazel" command target) " ")))
+   (mapconcat #'shell-quote-argument
+              (append bazel-build-bazel-command (list command target)) " ")))
 
-(defun bazel-build--read-target (prompt)
-  "Read a Bazel build target from the minibuffer.  PROMPT is a read-only prompt."
+(defun bazel-build--read-target (command)
+  "Read a Bazel build target from the minibuffer.
+COMMAND is a read-only Bazel command."
   (let* ((file-name (buffer-file-name))
          (workspace-root
           (or (bazel-util-workspace-root file-name)
               (user-error "Not in a Bazel workspace.  No WORKSPACE file found")))
          (package-name
           (or (bazel-util-package-name file-name workspace-root)
               (user-error "Not in a Bazel package.  No BUILD file found")))
-         (initial-input (concat "//" package-name)))
+         (initial-input (concat "//" package-name))
+         (prompt (concat (string-join bazel-build-bazel-command " ") " "
```

Comment:
I'd use combine-and-quote-strings here, it's meant for such a user-friendly display of command-line lists

Reply:
TIL again, thanks! Not super happy about the final `""` here but I don’t think there’s a good way to get a space between the prompt and the user input.